src/main/java/org/candlepin/subscriptions/db/SubscriptionRepository.java

```diff
@@ -49,4 +50,12 @@
     @Query("SELECT s FROM Subscription s where s.endDate > CURRENT_TIMESTAMP " +
         "AND s.subscriptionId = :subscriptionId")
     Optional<Subscription> findActiveSubscription(@Param("subscriptionId") String subscriptionId);
+
+    @Query("SELECT s FROM Subscription s WHERE s.accountNumber = :accountNumber AND " +
+        "s.sku = ALL (SELECT DISTINCT o.sku FROM Offering o WHERE " +
+        ":#{#key.usage} = o.usage AND " +
+        ":#{#key.sla} = o.serviceLevel AND " +
+        ":#{T(java.lang.Integer).parseInt(#key.productId)} MEMBER OF o.productIds)")
+    Optional<Subscription> findSubscriptionByAccountAndUsageKey(@Param("accountNumber") String accountNumber,
+        @Param("key") UsageCalculation.Key usageKey);
```

Comment:
I think we should find all matching subscriptions and then pick one in the logic. Example use case: Marketplace subscription is purchased for 1 year, then another is bought the next year. We need to select the appropriate one for that timeframe. Also in the future we may need to support multiple active, so may as well fetch all matches for now.